scripts/extract_pdf_abstract.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+# coding: utf8
+"""Extracts the abstracts from the proceedings PDFs.
+
+Usage
+-----
+
+$ python ./scripts/extrac_pdf_abstract.py \
+    ./path/to/pdfs
+
+"""
+import os
+import glob
+import io
+import tempfile
+import argparse
+import pdfminer.high_level
+import pdfminer.layout
+import pdfminer.settings
+from pdfrw import PdfReader, PdfWriter
+from pdfrw.findobjs import page_per_xobj
+pdfminer.settings.STRICT = False
+
+
+def extract_first_page(fname):
+
+    # create a temporary PDF file
+    path_tmpfile = os.path.join(tempfile.gettempdir(), os.path.basename(fname))
+
+    # extract all elements from the PDF and put them on separate pages
+    pages = list(page_per_xobj(PdfReader(fname).pages, margin=0.5*72))
+
+    # write to temp file
+    writer = PdfWriter(path_tmpfile)
+    writer.addpages(pages)
+    writer.write()
+
+    return path_tmpfile
+
+
+def extract_text(fname):
+    """Get all the text from the PDF."""
+
+    laparams = pdfminer.layout.LAParams()
+    for param in ('all_texts', 'detect_vertical', 'word_margin', 'char_margin', 'line_margin', 'boxes_flow'):
+        paramv = locals().get(param, None)
+        if paramv is not None:
+            setattr(laparams, param, paramv)
+
+    # send output to a string stream
+    outfp = io.StringIO()
+
+    with open(fname, 'rb') as fp:
+        pdfminer.high_level.extract_text_to_fp(fp, outfp=outfp, codec='utf-8',
+                                               laparams=laparams, pages=0)
+
+    return outfp.getvalue()
+
+
+def extract_abstract(raw_text):
+    """Search in the text for keywords and extract text in between."""
+
+    abs_index = raw_text.find('ABSTRACT')
+    intro_index = abs_index
+
+    intro_index = raw_text.find('1. INTRODUCTION')
+
+    if intro_index == -1:
+        intro_index = raw_text.find('1.  INTRODUCTION')
+
+    abstract = raw_text[abs_index + 8:intro_index]
+    abstract = abstract.strip()
+
+    # replace some ugly things
+    abstract = abstract.replace('-\n', '')
+    abstract = abstract.replace('\n', ' ')
+    abstract = abstract.replace('ﬁ', 'fi')
+
+    return abstract
+
+
+def main(path_pdfs):
+    """Main function which defines the processing pipeline."""
+
+    # loop through all PDFs (could be parallized via joblib)
+    for cur_fn_pdf in glob.glob(os.path.join(path_pdfs, '*.pdf')):
```

Comment:
I imagine some of this PDF parsing is time consuming ya? In the event that it is, we may want to architect this loop for easier parallelization. There are two ways I've seen this done:
- use `joblib` in python and wrap the inner loop in a `delayed` method
- let the Python script only do one file at a time and use something like `parallel` in bash, via something like `ls /my/pdf/path | parallel -j0 -v "./scripts/extract_pdf_abstract.py {} ./output/dir"`

pros and cons to both, and no need to do it now, but worth having on the radar

Reply:
Will do joblib.

Reply:
Joblib is in there, actually it's not too bad performance-wise.